ffi/src/repr_c.rs

```diff
@@ -82,76 +82,76 @@ pub trait COutPtr<S>: CType<S> {
 pub unsafe trait NonLocal<S>: COutPtr<S> {}
 
 /// Type that cannot be transmuted into a [`ReprC`] type
-pub trait NonTransmute {}
+pub trait Cloned {}
 
-impl<R: Ir> NonTransmute for &R where R::Type: NonTransmute {}
-impl<R> NonTransmute for Box<R> {}
-impl<R> NonTransmute for &[R] {}
-impl<R> NonTransmute for Vec<R> {}
-impl<R: Ir, const N: usize> NonTransmute for [R; N] where R::Type: NonTransmute {}
+impl<R: Ir> Cloned for &R where R::Type: Cloned {}
+impl<R> Cloned for Box<R> {}
+impl<R> Cloned for &[R] {}
+impl<R> Cloned for Vec<R> {}
+impl<R: Ir, const N: usize> Cloned for [R; N] where R::Type: Cloned {}
 
 // NOTE: `CType` cannot be implemented for `&mut T`
-impl<R: Ir<Type = S> + CType<S>, S: NonTransmute> CType<&S> for &R {
-    type ReprC = *const <R as CType<R::Type>>::ReprC;
+impl<R: CType<S>, S: Cloned> CType<&S> for &R {
+    type ReprC = *const R::ReprC;
 }
-impl<'itm, R: Ir<Type = S> + CTypeConvert<'itm, S, C> + Clone, S: NonTransmute, C: ReprC>
-    CTypeConvert<'itm, &S, *const C> for &'itm R
+impl<'itm, R: CTypeConvert<'itm, S, C> + Clone, S: Cloned, C: ReprC>
+    CTypeConvert<'itm, &'itm S, *const C> for &'itm R
 {
-    type RustStore = (Option<C>, <R as CTypeConvert<'itm, R::Type, C>>::RustStore);
-    type FfiStore = (Option<R>, <R as CTypeConvert<'itm, R::Type, C>>::FfiStore);
+    type RustStore = (Option<C>, R::RustStore);
+    type FfiStore = (Option<R>, R::FfiStore);
 
-    fn into_repr_c(self, store: &'itm mut Self::RustStore) -> *const C {
-        store.0.insert(self.clone().into_repr_c(&mut store.1))
+    fn lower(self, store: &'itm mut Self::RustStore) -> *const C {
```

Comment:
This is a little less clear than `into_repr_c`

Reply:
I agree